flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -303,6 +323,9 @@ bool CodeGenAction::beginSourceFileAction() {
   Fortran::parser::Program &parseTree{*ci.getParsing().parseTree()};
   lb.lower(parseTree, ci.getInvocation().getSemanticsContext());
 
+  // Add dependent libraries
+  addDepdendentLibs(*mlirModule, ci);
```

Comment:
What are the semantics of "dependent-lib"  - what's the consumer of this flag? Wondering whether this isn't too early in the compilation to call this hook.

Reply:
The consumer is that an LLVMIR dialect operation should be added to the MLIR module (in the global module scope) that will eventually get lowered by the standard MLIR LLVM dialect to LLVM ir transform. I don’t think it particularly matters where in the compilation flow that gets added as it doesn’t affect or get affected by any other transformation. I’m happy to add it later if you have a suggestion where it should be done, I mostly did it here because you still have all the information to construct the list of linker options without needing to pass any information further on.

Reply:
Apologies, I thought that this was [FrontendAction::beginSourceFile](https://github.com/banach-space/llvm-project/blob/d0b00b2c494b44fd1da7964b9334633215765c7f/flang/lib/Frontend/FrontendAction.cpp#L38) rather than `CodeGenAction::beginSourceFileAction()`.

I think that keeping it here is fine. Like you said, the ordering in this case doesn't matter that much.